fe/fe-core/src/main/java/org/apache/doris/catalog/TableIf.java

```diff
@@ -153,6 +156,80 @@ default int getBaseColumnIdxByName(String colName) {
 
     void write(DataOutput out) throws IOException;
 
+    default Map<String, Constraint> getConstraintMap() {
+        throw new RuntimeException("Not implemented constraint for table " + this.toString());
+    }
+
+    // Note this function is not thread safe
+    default void checkConstraintNotExistence(String name, Constraint primaryKeyConstraint,
+            Map<String, Constraint> constraintMap) {
+        if (constraintMap.containsKey(name)) {
+            throw new RuntimeException(String.format("Constraint name %s has existed", name));
+        }
+        for (Map.Entry<String, Constraint> entry : constraintMap.entrySet()) {
+            if (entry.getValue().equals(primaryKeyConstraint)) {
+                throw new RuntimeException(String.format(
+                        "Constraint %s has existed, named %s", primaryKeyConstraint, entry.getKey()));
+            }
+        }
+    }
+
+    default void addUniqueConstraint(String name, ImmutableList<Column> columns) {
+        writeLock();
+        try {
+            Map<String, Constraint> constraintMap = getConstraintMap();
```

Comment:
need to check repeat column and throw exception

Reply:
The set can handle repeat column case